keyring/key.go

```diff
@@ -0,0 +1,168 @@
+// keyring provides utilities for loading and using crypto keys in different formats
+package keyring
+
+import (
+	"crypto"
+	"crypto/ed25519"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+
+	"golang.org/x/crypto/ssh"
+
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+)
+
+type Key interface {
+	JWASignatureAlgorithm() (jwa.SignatureAlgorithm, error)
+	JWKThumbprintSHA256() (string, error)
+	JWTSerializer() (*jwt.Serializer, error)
+	SSHFingerprintSHA256() (string, error)
+}
+
+type keyImpl struct {
+	comment string
+	jwk     jwk.Key
+}
+
+// Open reads a key from file, returning a Key
+func Open(path string) (Key, error) {
+	// Read the key file
+	keyData, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read %v: %w", path, err)
+	}
+
+	return Parse(keyData)
+}
+
+// ParseString is a shortcut for calling Parse with a string
+func ParseString(s string) (Key, error) {
+	return Parse([]byte(s))
+}
+
+// Parse parsses raw key bytes
+func Parse(bytes []byte) (Key, error) {
+	// Try to parse the key as a JWK
+	if key, err := jwk.ParseKey(bytes); err == nil {
+		return &keyImpl{jwk: key}, nil
+	}
+
+	// Try to parse as PEM
+	if key, err := jwk.ParseKey(bytes, jwk.WithPEM(true)); err == nil {
```

Comment:
I'd think PEM would have sufficed (as this is a demo app) but hey, it's already there so OK

Reply:
Yes these functions were brought over from my work on nuts-jwt-generator so they support many formats